pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -896,37 +928,41 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 						volumeInfos,
 					)
 					if err != nil {
-						return err
+						continue
 					}
 				}
 			}
 		}
-		// In case Portworx if the snapshot ID is populated for every volume then the snapshot
-		// process is considered to be completed successfully.
-		// This ensures we don't execute the post-exec before all volume's snapshot is completed
-		for driverName := range pvcMappings {
-			var driver volume.Driver
-			driver, err = volume.Get(driverName)
+	}
+	// In case Portworx if the snapshot ID is populated for every volume then the snapshot
+	// process is considered to be completed successfully.
+	// This ensures we don't execute the post-exec before all volume's snapshot is completed
+	volumeInfosAll := make([]*stork_api.ApplicationBackupVolumeInfo, 0)
+	for driverName := range pvcMappings {
+		var driver volume.Driver
+		driver, err = volume.Get(driverName)
+		if err != nil {
+			return fmt.Errorf("error getting backup status: %v", err)
+		}
+		if driverName == volume.PortworxDriverName {
+			volumeInfos, err := driver.GetBackupStatus(backup)
+			volumeInfosAll = append(volumeInfosAll, volumeInfos...)
 			if err != nil {
-				return err
+				logrus.Errorf("error getting backup status: %v", err)
+				continue
 			}
-			if driverName == volume.PortworxDriverName {
-				volumeInfos, err := driver.GetBackupStatus(backup)
-				if err != nil {
-					return fmt.Errorf("error getting backup status: %v", err)
-				}
-				for _, volInfo := range volumeInfos {
-					if volInfo.BackupID == "" {
-						log.ApplicationBackupLog(backup).Infof("Snapshot of volume [%v] from namespace [%v] hasn't completed yet, retry checking status", volInfo.PersistentVolumeClaim, volInfo.Namespace)
-						// Some portworx volume snapshot is not completed yet
-						// hence we will retry checking the status in the next reconciler iteration
-						// *stork_api.ApplicationBackupVolumeInfo.Status is not being checked here
-						// since backpID confirms if the snapshot is done or not already
-						return nil
-					}
+			for _, volInfo := range volumeInfos {
+				if volInfo.BackupID == "" {
+					log.ApplicationBackupLog(backup).Infof("Snapshot of volume [%v] hasn't completed yet, retry checking status", volInfo.PersistentVolumeClaim)
```

Comment:
ny reason why we removed namespace? In case of same volumeName different namespace, this will have no help in debugging.